packages/components/src/components/CodeEditor.tsx

```diff
@@ -1,76 +1,201 @@
+import { defaultKeymap } from "@codemirror/commands";
+import { setDiagnostics } from "@codemirror/lint";
+
+// From basic setup
+import {
+  lineNumbers,
+  highlightActiveLineGutter,
+  highlightSpecialChars,
+  drawSelection,
+  dropCursor,
+  highlightActiveLine,
+} from "@codemirror/view";
+import { history, historyKeymap, indentWithTab } from "@codemirror/commands";
+import { highlightSelectionMatches, searchKeymap } from "@codemirror/search";
+import {
+  closeBrackets,
+  autocompletion,
+  closeBracketsKeymap,
+  completionKeymap,
+} from "@codemirror/autocomplete";
+import { lintKeymap } from "@codemirror/lint";
+import {
+  foldGutter,
+  indentOnInput,
+  bracketMatching,
+  foldKeymap,
+} from "@codemirror/language";
+import { lightThemeHighlightingStyle } from "./languageSupport/highlightingStyle";
+
 interface CodeEditorProps {
   value: string;
   onChange: (value: string) => void;
   onSubmit?: () => void;
   oneLine?: boolean;
   width?: number;
-  height: number;
+  height?: number;
   showGutter?: boolean;
   errorLocations?: SqLocation[];
 }
 
+const compTheme = new Compartment();
+const compGutter = new Compartment();
+const compUpdateListener = new Compartment();
+const compSubmitListener = new Compartment();
+
 export const CodeEditor: FC<CodeEditorProps> = ({
   value,
   onChange,
   onSubmit,
+  width,
   height,
   oneLine = false,
   showGutter = false,
   errorLocations = [],
 }) => {
-  const lineCount = value.split("\n").length;
-  const id = useMemo(() => _.uniqueId(), []);
-
-  // this is necessary because AceEditor binds commands on mount, see https://github.com/securingsincity/react-ace/issues/684
-  const onSubmitRef = useRef<typeof onSubmit | null>(null);
-  onSubmitRef.current = onSubmit;
-
-  const editorEl = useRef<AceEditor | null>(null);
-
-  return (
-    <AceEditor
-      ref={editorEl}
-      value={value}
-      mode="golang"
-      theme="github"
-      width="100%"
-      fontSize={14}
-      height={String(height) + "px"}
-      minLines={oneLine ? lineCount : undefined}
-      maxLines={oneLine ? lineCount : undefined}
-      showGutter={showGutter}
-      highlightActiveLine={false}
-      showPrintMargin={false}
-      onChange={onChange}
-      name={id}
-      editorProps={{
-        $blockScrolling: true,
-      }}
-      setOptions={{}}
-      commands={[
-        {
-          name: "submit",
-          bindKey: { mac: "Cmd-Enter", win: "Ctrl-Enter" },
-          exec: () => onSubmitRef.current?.(),
-        },
-      ]}
-      markers={errorLocations?.map((location) => ({
-        startRow: location.start.line - 1,
-        startCol: location.start.column - 1,
-        endRow: location.end.line - 1,
-        endCol: location.end.column - 1,
-        className: "ace-error-marker",
-        type: "text",
-      }))}
-    />
+  const editor = useRef<HTMLDivElement>(null);
+  const editorView = useRef<EditorView | null>(null);
+  const languageSupport = useMemo(squiggle, []);
```

Comment:
`squiggle` is constant, why is this necessary? You could simply import it as `languageSupport`.

Reply:
Yeah, I've added it from the start to support dialect changes later on, but didn't require it in the end)